tests/lib/Server/Output/ValueObjectVisitor/LocationTest.php

```diff
@@ -0,0 +1,154 @@
+use eZ\Publish\Core\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\Repository\Values\ContentType\ContentType;
+use EzSystems\EzPlatformRest\Tests\Output\ValueObjectVisitorBaseTest;
+use EzSystems\EzPlatformRest\Server\Output\ValueObjectVisitor;
+use PHPUnit\Framework\MockObject\MockObject;
+
+final class LocationTest extends ValueObjectVisitorBaseTest
+{
+    private const MAIN_LOCATION_ID = 78;
+
+    private const UNAUTHORIZED_MAIN_LOCATION_ID = 111;
+
+    private const LOCATION_ID = 55;
+
+    /** @var \eZ\Publish\API\Repository\LocationService|MockObject */
+    private $locationServiceMock;
+
+    /** @var ContentService|MockObject */
+    private $contentServiceMock;
+
+    protected function setUp(): void
+    {
+        $this->permissionResolverMock = $this->createMock(PermissionResolver::class);
+        $this->locationServiceMock = $this->createMock(LocationService::class);
+        $this->contentServiceMock = $this->createMock(ContentService::class);
+
+        parent::setUp();
+    }
+
+    /**
+     * @dataProvider providerForMainLocationId
+     */
+    public function testVisitLocationWithDifferentContentMainLocations(
+        ?int $mainLocationId,
+        int $locationId
+    ): void {
+        $visitor = $this->getVisitor();
+        $generator = $this->getGenerator();
+
+        $generator->startDocument(null);
+
+        $contentId = 7;
+        $versionInfo = new VersionInfo();
+
+        $location = new Location([
+            'id' => $locationId,
+            'path' => ['1', '25', '42'],
+            'priority' => 1,
+            'sortField' => ApiLocation::SORT_FIELD_DEPTH,
+            'sortOrder' => ApiLocation::SORT_ORDER_ASC,
+            'parentLocationId' => 42,
+            'contentInfo' => new ContentInfo([
+                'id' => $contentId,
+                'mainLocationId' => $mainLocationId,
+            ]),
+            'content' => new Content([
+                'id' => $contentId,
+                'contentType' => new ContentType(),
+                'versionInfo' => $versionInfo,
+            ]),
+        ]);
+
+        $this->mockLoadLocation($location);
+
+        $this->contentServiceMock->expects(self::once())
+            ->method('loadRelations')
+            ->with($versionInfo)
+            ->willReturn([]);
+
+        $visitor->visit(
+            $this->getVisitorMock(),
+            $generator,
+            $location
+        );
+
+        $result = $generator->endDocument(null);
+
+        self::assertNotNull($result);
+
+        $this->assertXMLTag(
+            [
+                'tag' => 'Location',
+            ],
+            $result,
+            'Invalid <Location> element.',
+        );
+
+        $this->assertXMLTag(
+            [
+                'tag' => 'Location',
+                'content' => $location->id . 1 . 'false' . 'false',
+            ],
+            $result,
+            'Invalid <Location> attributes.',
+        );
```

Comment:
Why do you need 2x `assertXMLTag`?
How does the actual XML structure look like?
```php
$location->id . 1 . 'false' . 'false',
```
looks a bit suspicious.

Reply:
> Why do you need 2x assertXMLTag?
Bad copy-paste.

> How does the actual XML structure look like?
```
<?xml version="1.0" encoding="UTF-8"?>\n
<Location media-type="application/vnd.ez.api.Location+xml" href=""><id>55</id><priority>1</priority><hidden>false</hidden><invisible>false</invisible><ParentLocation media-type="application/vnd.ez.api.Location+xml" href=""/><pathStr
ing></pathString><depth></depth><childCount>0</childCount><remoteId></remoteId><Children media-type="application/vnd.ez.api.LocationList+xml" href=""/><Content media-type="application/vnd.ez.api.Content+xml" href=""/><sortField>DEPT
H</sortField><sortOrder>ASC</sortOrder><UrlAliases media-type="application/vnd.ez.api.UrlAliasRefList+xml" href=""/><ContentInfo media-type="application/vnd.ez.api.ContentInfo+xml" href=""/></Location>
```
`content` is basically content between XML tags.